src/main/java/teammates/ui/controller/AdminInstructorAccountAddAction.java

```diff
@@ -297,37 +298,24 @@ private String getDemoCourseIdRoot(String instructorEmail){
     private String generateNextDemoCourseId(String instructorEmailOrProposedCourseId, int maximumIdLength){
         final boolean isFirstCourseId = instructorEmailOrProposedCourseId.contains("@");
         if(isFirstCourseId){
-            return trimCourseIdToMaximumLengthIfNecessary(getDemoCourseIdRoot(instructorEmailOrProposedCourseId)
-                    , maximumIdLength);
+            return StringHelper.truncateHead(getDemoCourseIdRoot(instructorEmailOrProposedCourseId)
+                                            , maximumIdLength);
         } else {
             final boolean isFirstTimeDuplicate = instructorEmailOrProposedCourseId.endsWith("-demo"); 
             if(isFirstTimeDuplicate){
-                return trimCourseIdToMaximumLengthIfNecessary(instructorEmailOrProposedCourseId + "0"
-                        , maximumIdLength);
+                return StringHelper.truncateHead(instructorEmailOrProposedCourseId + "0"
+                                                , maximumIdLength);
             } else {
                 final int lastIndexOfDemo = instructorEmailOrProposedCourseId.lastIndexOf("-demo");
                 final String root = instructorEmailOrProposedCourseId.substring(0, lastIndexOfDemo);
                 final int previousDedupSuffix = Integer.parseInt(instructorEmailOrProposedCourseId.substring(lastIndexOfDemo + 5));
-                
-                return trimCourseIdToMaximumLengthIfNecessary(root + "-demo" + (previousDedupSuffix+1)
-                        , maximumIdLength);
+
+                return StringHelper.truncateHead(root + "-demo" + (previousDedupSuffix+1)
```

Comment:
This is not your change but could you add a space between operands and the operator `+` here to make the code look more consistent?